src/terminal/select.c

```diff
@@ -360,16 +360,19 @@ void guac_terminal_select_end(guac_terminal* terminal) {
     else {
 
         /* Store first row */
-        guac_terminal_clipboard_append_row(terminal, start_row, start_col, -1);
+        guac_terminal_buffer_row* buffer_row = guac_terminal_buffer_get_row(terminal->buffer, start_row, 0);
+        guac_terminal_clipboard_append_row(terminal, start_row, start_col, buffer_row->length - 2);
```

Comment:
1. `buffer_row->length` is not guaranteed to be at least 2 here unless the call to `guac_terminal_buffer_get_row()` requests this.
2. Why are we universally skipping the last two characters?

Reply:
My purpose it to omit the line break when an extra long line of text is separated because of insufficient length of the terminal simulator. I managed to use buffer_row->length - 1 with server version 1.3.0, but have to cut down another character with 1.4.0. Currently, the horizontal index of bound is length - 2, I'll figure it out with annotations in next commit.